### ISSUE TITLE
can we use guzzle 6?

### ISSUE THREAD

**Issue by cordoval** [constantcontact/php-sdk#113]

**Comment by chrisvanderpoel** [comment]
Agreed, a version that's compatible with Guzzle 6 would be very much appreciated.

**Comment by WoogieNoogie** [comment]
I'll take a look at making this conversion. It looks fairly straightforward, but drastically changes how requests are built, so I'll want to make sure I'm doing it in the best way possible.

**Comment by siljegovice** [comment]
When will we have it compatible with new Guzzle version?

**Comment by WoogieNoogie** [comment]
The biggest hurdle with this is Guzzle's requirement for a minimum of PHP 5.5 - dropping support for 5.3 was a hurdle in itself, and multiple versions for different minimums makes things difficult. I'll likely push out a non-standard release version for anyone that wants to sue Guzzle 6, which will also have a minimum of PHP 5.5. I don't have a timeline for this, but hopefully soon.

**Comment by cordoval** [comment]
I am no longer interested here so unsubscribing myself from this for the moment.

**Comment by WoogieNoogie** [comment]
PR #118 will take care of this. This change is fairly major, so we don't have a timeframe for releasing an official version (likely version 3.0.0), but you can install this via composer using the version "constantcontact/constantcontact" : "dev-development" once this PR is merged.

**Comment by JesseLeite** [comment]
@WoogieNoogie We're blocked at work here without Guzzle 6 support.  Our app is built on Laravel, which requires me to pull in Guzzle `"~5.3|~6.0"` for it's mail functionality.  I tried pulling `dev-development`, but then I get invalid action_by param error when trying to do a simple `addContact()`.

[There's a full explanation in my forum post here.](https://community.constantcontact.com/t5/Developer-Support-ask-questions/Bad-request-when-trying-to-addContact/m-p/275962#M9954)  In the thread there, Elijah mentions this as a bug in `dev-development` and recommends I use `dev-master` or `2.1.*` ...but again dependency conflict around Guzzle :(

**Comment by claytonfbell** [comment]
It would be really nice if Guzzle 6 could be supported. Or if either 5 or 6 was acceptable `~5.0|~6.0`.
My project has another library that requires `~6.0`.

**Comment by shannon7wallace** [comment]
Hi Clayton, I believe the dev branch of this does use guzzle 6. https://github.com/constantcontact/php-sdk/tree/development

Let me know if that's not the case. Or, use the v1-master branch, which is self-contained and doesn't use guzzle at all (just fork and pull, or download the zip).

**Comment by claytonfbell** [comment]
Thanks, I am using `dev-development`, it is working well.

```
{ 
  "require": {
    "constantcontact/constantcontact": "dev-development",
    ...
  }
}
```

**Comment by huebs** [comment]
@WoogieNoogie, is there a plan to tag a release with Guzzle 6 support anytime soon?

**Comment by huebs** [comment]
@shannon7wallace do you know when the 3.0.0 version with Guzzle 6 support is expected to be released?

**Comment by shannon7wallace** [comment]
Hi Huebs,

I'm pretty sure that Guzzle 6 is already supported in the dev branch of the SDK. Are you sure it's not?

If not, no, I don't have a timeline for it. Could we get you on the v1-master branch of the SDK, which is totally independent of Guzzle, if you're having a compatibility issue?

-Shannon

**Comment by huebs** [comment]
@shannon7wallace I'm aware the [`dev-development`](https://github.com/constantcontact/php-sdk/tree/development) branch has been updated to support Guzzle 6. However, it's considered bad practice to use [unbound version constraints](https://getcomposer.org/doc/faqs/why-are-unbound-version-constraints-a-bad-idea.md). It would be nice if we could at least get a version tagged v3.0.0-alpha (or something along those lines).

It would also be helpful to know a the tentative release schedule is for v3.0.0 in order to be able to know when we can expect to be able to depend on Guzzle 6 and/or other dependencies that require it in production code.

What is holding up the release? Is the implementation incomplete or does it just need to be more thoroughly tested? Is there anything the community can do to help?

**Comment by shannon7wallace** [comment]
I see, thanks for the explanation. There are a couple of reasons for the lack of a timeline.
1. We don't have a dedicated resource to do this. WoogieNoogie has totally different responsibilities here and does this out of the kindness of his heart. (Thanks Nick!)
2. We're building our V3 API, due out later this year.

On the bright side, I heard from WoogieNoogie just now, and he says he'll work on this request today. If you don't see some action by early next week, feel free to reach out. I'll probably either advise that yes, we need community help to make this happen, or that Nick has made progress but isn't quite there yet.

-Shannon

**Comment by WoogieNoogie** [comment]
The biggest reason for not completely jumping over to support Guzzle 6 is their constraint on a minimum of PHP 5.5. We're not comfortable dropping official support of 5.4 right now, and maintaining two different code bases due to two different major versions of Guzzle takes a bit more time than we really have to give right now.

I have tagged a prerelease/unstable version for 3.0.0-alpha. I'd like to hold off on considering it stable, likely until we move forward with the V3 API, and all of the endpoints and functionality will be moved over as well, but you're free to report and/or fix issues on the branch, and I'll attempt to be responsive to getting the changes merged in and cutting releases.

**Comment by giladmatko** [comment]
Hi @WoogieNoogie.
Do you have any idea when the 3.0.0-alpha will be released and stable to use?

Thanks.

**Comment by WoogieNoogie** [comment]
I'm pretty confident in the 3.0.0-alpha version, but we're still not considering it the "official" supported version until we're ready to drop support for PHP 5.4. However, if you would like to start using it, I'll be happy to answer any issues you have with it, and fix any bugs.

**Comment by moebaca** [comment]
Thanks @WoogieNoogie

**Comment by MACscr** [comment]
Still no support for it? I cant install the api because of guzzle dependency conflicts with the other api's i use.

**Comment by Zaskoda** [comment]
How is this still an issue?

**Comment by brmullikin** [comment]
So any update on this? Guzzle 5.3 doesn't work well with PHP 7 -- and PHP 5.6 has security end of life this December.

edit: using 3.0.0-alpha for now. Seems to work fine.
